external-crates/move/crates/move-transactional-test-runner/src/vm_test_harness.rs

```diff
@@ -87,24 +83,30 @@ impl Linkage {
                 .add_entry(*runtime_id, *storage_id)
                 .map_err(|err| err.finish(Location::Undefined))?;
         }
-        // Remap the root address if it is in the linkage table since it may be overriden
-        existing_linkage.root_package = existing_linkage
-            .linkage_table
-            .get(&existing_linkage.root_package)
-            .unwrap_or(&existing_linkage.root_package)
-            .clone();
         Ok(existing_linkage)
     }
 }
 
 impl PublishLinkageArgs {
     pub fn overlay(&self, existing_linkage: LinkageContext) -> VMResult<LinkageContext> {
-        let mut linkage = self.linkage.overlay(existing_linkage)?;
-        if let Some(location) = self.location {
-            linkage.root_package = location;
-        }
+        self.linkage.overlay(existing_linkage)
+    }
 
-        Ok(linkage)
+    pub fn resolve_publication_location(
+        &self,
+        sender: AccountAddress,
+        linkage: &LinkageContext,
+    ) -> AccountAddress {
+        // 1. Use location if specified; otherwise
+        // 2. Use the sender address (remapped if it exists in the linkage table); otherwise
+        // 3. Use the sender address
+        self.location.unwrap_or(
+            linkage
+                .linkage_table
+                .get(&sender)
+                .copied()
+                .unwrap_or(sender),
+        )
```

Comment:
This seems dubious: if this package is not in the linkage, shouldn't it be an error?